app/src/main/java/org/jellyfin/mobile/webapp/ConnectionHelper.kt

```diff
@@ -172,45 +165,26 @@ class ConnectionHelper(private val activity: MainActivity) : KoinComponent {
             // Set API client address
             apiClient.ChangeServerLocation(httpUrl.toString())
 
-            // TODO: resolve system info via API client
-            serverInfoResponse = fetchServerInfo(activity.httpClient, httpUrl)
-            if (serverInfoResponse != null)
+            serverInfo = apiClient.getPublicSystemInfo()
+            if (serverInfo != null)
                 break@loop
         }
 
-        if (httpUrl == null || serverInfoResponse == null) {
+        if (httpUrl == null || serverInfo == null) {
             showConnectionError()
             return null
         }
 
-        val isValidInstance = try {
-            val serverInfo = JSONObject(serverInfoResponse)
-            val version = serverInfo.getString("Version")
-                .split('.')
-                .mapNotNull(String::toIntOrNull)
-            when {
-                version.size != 3 -> false
-                version[0] == 10 && version[1] < 3 -> true // Valid old version
-                else -> serverInfo.getString("ProductName") == "Jellyfin Server"
-            }
-        } catch (e: JSONException) {
-            Timber.e(e, "Cannot get server info")
-            false
+        val version = serverInfo.version
+            .split('.')
+            .mapNotNull(String::toIntOrNull)
+
+        val isValidInstance = when {
+            version.size != 3 -> false
+            version[0] == 10 && version[1] < 3 -> true // Valid old version
```

Comment:
We should put the supported version in a constant somewhere instead of hardcoded in the source